contracts/dnssec-oracle/DNSSECImpl.sol

```diff
@@ -92,304 +72,49 @@ contract DNSSECImpl is DNSSEC, Owned {
-        }
-        return false;
-    }
-
-    /**
-     * @dev Returns data about the RRs (if any) known to this oracle with the provided type and name.
-     * @param dnstype The DNS record type to query.
-     * @param name The name to query, in DNS label-sequence format.
-     * @return inception The unix timestamp (wrapped) at which the signature for this RRSET was created.
-     * @return expiration The unix timestamp (wrapped) at which the signature for this RRSET expires.
-     * @return hash The hash of the RRset.
-     */
-    function rrdata(uint16 dnstype, bytes calldata name) external override view returns (uint32, uint32, bytes20) {
-        RRSet storage result = rrsets[keccak256(name)][dnstype];
-        return (result.inception, result.expiration, result.hash);
-    }
-
-    function _submitRRSet(RRSetWithSignature memory input, bytes memory proof) internal returns (bytes memory) {
-        RRUtils.SignedSet memory rrset;
-        rrset = validateSignedSet(input, proof);
-
-        RRSet storage storedSet = rrsets[keccak256(rrset.name)][rrset.typeCovered];
-        if (storedSet.hash != bytes20(0)) {
-            // To replace an existing rrset, the signature must be at least as new
-            require(RRUtils.serialNumberGte(rrset.inception, storedSet.inception));
-        }
-        rrsets[keccak256(rrset.name)][rrset.typeCovered] = RRSet({
-            inception: rrset.inception,
-            expiration: rrset.expiration,
-            hash: bytes20(keccak256(rrset.data))
-        });
-
-        emit RRSetUpdated(rrset.name, rrset.data);
-
-        return rrset.data;
-    }
-
-    /**
-     * @dev Submits a signed set of RRs to the oracle.
-     *
-     * RRSETs are only accepted if they are signed with a key that is already
-     * trusted, or if they are self-signed, and the signing key is identified by
-     * a DS record that is already trusted.
+     * @dev Validates an RRSet against the already trusted RR provided in `proof`.
      *
      * @param input The signed RR set. This is in the format described in section
      *        5.3.2 of RFC4035: The RRDATA section from the RRSIG without the signature
      *        data, followed by a series of canonicalised RR records that the signature
      *        applies to.
-     * @param proof The DNSKEY or DS to validate the signature against. Must Already
-     *        have been submitted and proved previously.
+     * @param proof The DNSKEY or DS to validate the signature against.
+     * @param now The current timestamp.
      */
-    function validateSignedSet(RRSetWithSignature memory input, bytes memory proof) internal view returns(RRUtils.SignedSet memory rrset) {
+    function validateSignedSet(RRSetWithSignature memory input, bytes memory proof, uint256 now) internal view returns(RRUtils.SignedSet memory rrset) {
         rrset = input.rrset.readSignedSet();
-        require(validProof(rrset.signerName, proof));
```

Comment:
Why do we not need this line anymore?

Reply:
`validProof` looked up the proof in the internal DB of already-validated RRSets. Since we no longer have this, we instead check that the chain of proofs is valid in the verification functions further down.